qiskit/opflow/gradients/natural_gradient.py

```diff
@@ -120,43 +113,71 @@ def convert(
         # Instantiate the QFI metric which is used to re-scale the gradient
         metric = self._qfi_method.convert(operator[-1], params) * 0.25
 
-        # Define the function which compute the natural gradient from the gradient and the QFI.
         def combo_fn(x):
-            c = np.real(x[0])
-            a = np.real(x[1])
-            if self.regularization:
-                # If a regularization method is chosen then use a regularized solver to
-                # construct the natural gradient.
-                nat_grad = NaturalGradient._regularized_sle_solver(
-                    a, c, regularization=self.regularization
-                )
-            else:
-                try:
-                    # Try to solve the system of linear equations Ax = C.
-                    nat_grad = np.linalg.solve(a, c)
-                except np.linalg.LinAlgError:  # singular matrix
-                    nat_grad = np.linalg.lstsq(a, c)[0]
-            return np.real(nat_grad)
+            return self.nat_grad_combo_fn(x, self.regularization)
 
         # Define the ListOp which combines the gradient and the QFI according to the combination
         # function defined above.
         return ListOp([grad, metric], combo_fn=combo_fn)
 
+    @staticmethod
+    def nat_grad_combo_fn(x, regularization=None):
+        """
+        Natural Gradient Function Implementation
+        Args:
+            x: Iterable consisting of Gradient, Quantum Geometric Tensor
+            regularization: Regularization method
+
+        Returns: Natural Gradient
+
+        """
+        c = x[0]
+        a = x[1]
+        if any(np.abs(np.imag(c_item)) > 1e-8 for c_item in c):
+            raise Warning("The imaginary part of the gradient are non-negligible.")
+        if np.any([[np.abs(np.imag(a_item)) > 1e-8 for a_item in a_row] for a_row in a]):
+            raise Warning("The imaginary part of the gradient are non-negligible.")
+        c = np.real(c)
+        a = np.real(a)
+
+        if regularization:
+            # If a regularization method is chosen then use a regularized solver to
+            # construct the natural gradient.
+            nat_grad = NaturalGradient._regularized_sle_solver(a, c, regularization=regularization)
+        else:
+            # Check if numerical instabilities lead to a metric which is not positive semidefinite
+            while True:
```

Comment:
This seems like it could add some overhead that we didn't have before. Is this really necessary or can we give the user the responsibility that the inputs make sense? 

If we really do need it, could we maybe move this into the gradients as a check or make it optional? And the `while True` makes me a bit nervous 😛 are we certain that this will always terminate?